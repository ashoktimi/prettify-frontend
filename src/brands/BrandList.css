/* Bandslist style for navbar  */
.BrandList-column {
  display: flex;
  justify-content:space-around;
  position: absolute;
  flex-wrap: wrap;
  left: 0;
  right: 0;   
  margin: auto;
  list-style: none;
  background-color: rgb(195, 206, 182);
  padding:10%;
  z-index: 2;
}

.BrandList-li{
  margin-bottom: 1.5rem;  
  margin-right: 2rem;
}
.BrandList-link {
  color: black;
  text-decoration: none;
  text-transform: capitalize;
  font-family: 'Heebo', sans-serif;
  font-size: 25px;
}
  
.BrandList-link:hover {
  text-decoration: underline;
}

/* Bandslist style for toggler  */

.Brandlist-toggler-ul {
  list-style: none;
}

.Brandlist-toggler-ul li {
  margin: 2.3rem 0;
  border-bottom: .5px solid rgb(233, 227, 227);
}
.Brandlist-toggler-ul a {
  text-transform: capitalize;
  font-weight: bold;
  text-decoration: none; 
  color: #333;
}
.Brandlist-toggler-ul a:hover{
  color: purple;
}

  
