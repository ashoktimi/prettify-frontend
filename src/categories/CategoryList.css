/* css property for categorylist in navbar */
.CategoryList-column {
  display: flex;
  justify-content: center;
  position: absolute;
  flex-wrap: wrap;
  left: 0;
  right: 0;   
  margin: auto;
  list-style: none;
  background-color: rgb(195, 206, 182);
  padding:10%;
  z-index: 2;
}
.CategoryList-li{
  margin-bottom: 1.5rem;  
  margin-right: 3rem;
}
  .CategoryList-link {
    color: black;
    text-decoration: none;
    text-transform: capitalize;
    font-family: 'Heebo', sans-serif;
    font-size: 25px;
}
    
.BrandList-link:hover {
  text-decoration: underline;
}

/* css property for categorylist in offcanvas toggler */

.Categorylist-toggler-ul {
  list-style: none;
}

.Categorylist-toggler-ul li {
  margin: 3.7rem 0;
  border-bottom: .5px solid rgb(233, 227, 227);
}
.Categorylist-toggler-ul a {
  text-transform: capitalize;
  font-weight: bold;
  text-decoration: none; 
  color: #333;
}
.Categorylist-toggler-ul a:hover{
  color: purple;
}