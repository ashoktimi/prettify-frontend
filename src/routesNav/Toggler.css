.NavBar-name-div{
    margin: auto;
    left: 0;
    right: 0;
}
.NavBar-name{
    font-family: 'Caveat', cursive;
    color: purple;
    letter-spacing: 0.5rem;
    font-size: 3rem;
    text-decoration: none;
}
.NavBar-name:hover{
  color: purple;
}
.navbar-toggler{
  color:rgb(243, 224, 230);
  border: none;
}

.NavBar-auth_Btn{
  margin-left: 1rem;
  color: #333333;
  text-decoration: none;
  font-weight: bold;
  font-family: 'Heebo', sans-serif;
} 
.Toggler-logout{
  margin-right: 10rem;
  color: #333333;
  text-decoration: none;
  font-weight: bold;
  font-family: 'Heebo', sans-serif;
  margin-left: 2.5rem;
  margin-top: 10px;
  font-size: 20px;
} 
.Toggler-logout:hover{
  color: black;
}

.Toggler-bag{
  color:black;
  text-decoration: none;
  margin-left: 2.5rem;
  margin-bottom: 1.5rem;
}
.Toggler-bag:hover{
  color: black;
}



/* for toggler elements */
.Toggler-h3{
  left: 0;
  right: 0;
  margin: auto;
  color: black;
  font-size: 2rem;
}

.Toggler-ul {
  list-style: none;
}

.Toggler-ul li {
  margin: 3rem 0;
  border-bottom: .5px solid rgb(233, 227, 227);
}

.Toggler-ul a {
  font-weight: bold;
  text-decoration: none;  
  color: #333;
}
.Toggler-ul a:hover {
  color: #9c6a8d;
}
