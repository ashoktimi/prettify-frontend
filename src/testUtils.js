import React from "react";
import UserContext from "./auth/userContext";


const demoUser = {
  username: "testuser",
  first_name: "testfirst",
  last_name: "testlast",
  email: "test@test.net",
  photo_url: null,
};

const UserProvider =
    ({ children, currentUser = demoUser, handleToggleOffcanvas = () => false }) => (
    <UserContext.Provider value={{ currentUser, handleToggleOffcanvas }}>
      {children}
    </UserContext.Provider>
);

function setup(){
  window.matchMedia = jest.fn().mockImplementation(query => {
    return {
      matches: false,
      media: query,
      onchange: null,
      addListener: jest.fn(),
      removeListener: jest.fn(),
      addEventListener: jest.fn(),
      removeEventListener: jest.fn(),
      dispatchEvent: jest.fn(),
    };
  });
}

function teardown(){
   window.matchMedia.mockClear();
   delete window.matchMedia;
}

function commonBeforeEach() {
  setup();
}

function commonAfterEach() {
  teardown();
}

export { UserProvider, commonBeforeEach, commonAfterEach };