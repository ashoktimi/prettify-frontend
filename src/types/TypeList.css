  .TypeList{
    margin-top: 10rem;
  }
  .TypeLists{
    margin-top: 10rem;
  }
  .TypeList-link {
    display:block;
    text-decoration: none;
    text-transform: capitalize;
    font-family: 'Heebo', sans-serif;
    font-weight: bold;
    color: #333;
    margin: 4.7rem 2rem;
    border-bottom: .5px solid rgb(233, 227, 227);    
  }
    
  .TypeList-link:hover {
    color: purple;
  }

  .list {
    display: none;
  }
  
  .element:hover .list {
    display: block;
  }
