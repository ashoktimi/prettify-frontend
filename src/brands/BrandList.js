import React from "react";
import { Link } from "react-router-dom";
import './BrandList.css'

/** Show page with list of brands.
 *
 * On mount, loads brands from API. 
 *
 * This is routed to at /brands
 */
const BrandList = ({ brands, handleHover }) =>{
    return (
      <>
      {brands && handleHover ?
        <div >
          <ul className="BrandList-column">
            {brands.map(b => (            
              <li className="BrandList-li" key={b.id} onClick={handleHover}>
                 <Link className="BrandList-link" to={`/brands/${(b.name).trim().replace(/ /g, "-")}/${b.id}`}>
                   {b.name}
                 </Link>  
              </li>       
            ))}
          </ul> 
        </div>
        :
        <ul className="Brandlist-toggler-ul">
          {brands.map(brand => (
          <li key={brand.id}><Link to={`/brands/${brand.name}/${brand.id}`}>{brand.name}</Link></li>
          ))}
        </ul>  
      }
      </>
    );
}

export default  BrandList;


