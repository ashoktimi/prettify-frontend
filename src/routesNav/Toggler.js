import React, {useState, useContext, useEffect} from "react";
import { Link } from 'react-router-dom';
import PrettifyApi from "../api/api";
import BrandList from "../brands/BrandList";
import CategoryList from "../categories/CategoyList";
import TagList from "../tags/TagLIst";
import TypeList from "../types/TypeList";
import UserContext  from "../auth/userContext";
import 'bootstrap/dist/css/bootstrap.css';
import { AiOutlineShoppingCart } from 'react-icons/ai';
import { Container, Navbar, Offcanvas } from 'react-bootstrap'
import './Toggler.css';




const Toggler = ({ logout, brands, categories, tags }) =>{

  const { currentUser, handleToggleOffcanvas} = useContext(UserContext);
  const [showBrands, setShowBrands] = useState(false);
  const [showCategories, setShowCategories] = useState(false);
  const [showTags, setShowTags] = useState(false);
  const [type, setType] = useState([]);
  const [showTypes, setShowTypes] = useState(false);
  const [value, setValue] = useState('nail'); 
  const [typeDetail, setTypeDetail] = useState([]);


  
  useEffect(() =>{
    async function getBrands() {
    let types = await PrettifyApi.getAllTypes();     
    setType(types.types);
    }
    getBrands()   
  },[])

  const uniqueTypes = [...new Set(type.map(t => t.type))];

  useEffect(() => {
    async function fetchTypeDetail() {
      if (value) {
        const response = await PrettifyApi.getTypes(value);
        setTypeDetail(response.types);
      }
    }
    fetchTypeDetail();
  }, [value]);
  
  function handleTypeClick(event) {
    const key = event.target.getAttribute('data-key');
    setValue(key);
    setShowTypes(true);
  }

  const handleBrandsClick = () => {
    setShowBrands(true);
  };
  const handleCateoryClick = () => {
    setShowCategories(true);
  };
  const handleTagClick = () => {
    setShowTags(true);
  };
  const handleBackClick = () => {
    setShowBrands(false);
    setShowCategories(false);
    setShowTags(false);
    setShowTypes(false);
  };

    return(
        <>  
        <Navbar expand='bg' style={{padding:0}}>
          <Container fluid>
            <Navbar.Toggle aria-controls={`offcanvasNavbar-expand`}/>
            <div className="NavBar-name-div">
              {currentUser ? 
                <Link to="/products" className="NavBar-name">Prettify</Link> 
                :               
                <Link to="/" className="NavBar-name" style={{marginLeft: "10rem"}}>Prettify</Link>
              }
            </div>
            {currentUser ?
            <>
            </>
             :
            <div style={{marginRight: '5rem'}}>
              <Link className="NavBar-auth_Btn" to="/login">Login</Link>  
              <Link className="NavBar-auth_Btn" to="/signup">Signup</Link>
             </div>
             }
            <Navbar.Offcanvas
              id={`offcanvasNavbar-expand`}
              aria-labelledby={`offcanvasNavbarLabel-expand`}
              placement="start"
            >
            {currentUser ?
              <>
              <Offcanvas.Header closeButton>
                {(brands.length > 0 || categories.length || tags.length> 0 || typeDetail.length > 0) && (showBrands || showCategories || showTags || showTypes)&& (
                  <button  onClick={handleBackClick} style={{border:'none', fontSize:'25px', backgroundColor:'white'}}> {`<`} </button>
                )}
                <Offcanvas.Title id={`offcanvasNavbarLabel-expand`} style={{color:'purple', left:0, right:0, margin:'auto'}}>
                  Prettify
                </Offcanvas.Title>
              </Offcanvas.Header>
              <Offcanvas.Body>
              
              {brands.length > 0  && showBrands ? (    
               <BrandList brands={brands} />
              ) : (categories.length > 0 && showCategories) ? (
               <CategoryList categories={categories} />
              ): (tags.length > 0 && showTags) ?(
               <TagList tags={tags}/>
              ):(typeDetail.length > 0 && showTypes)?(
               <TypeList types={typeDetail}/>
              ):

              <ul className="Toggler-ul" >
                <li><Link to="/products"  >ALL</Link></li>
                {uniqueTypes.map(type => (
                  <li key={type}>
                    <Link data-key={type} onClick={handleTypeClick}>{type.toUpperCase()}</Link>
                  </li>
                ))}
                <li><Link onClick={handleBrandsClick}> BRANDS </Link></li>
                <li><Link onClick={handleCateoryClick} >CATEGORY</Link></li>
                <li><Link onClick={handleTagClick} >TAGS</Link></li>
              </ul>             
              }
              </Offcanvas.Body>
              <Link className="Toggler-bag" onClick={handleToggleOffcanvas}>
                <AiOutlineShoppingCart size={30}/>
              </Link>
              <Link  to="/" className="Toggler-logout" onClick={logout} style={{marginBottom:'35px'}} >Log out</Link>      
              </>
              : <h3 className="Toggler-h3">Please login to see.</h3>}
            </Navbar.Offcanvas>
          </Container>
        </Navbar>
      </>
    )
}

export default Toggler;