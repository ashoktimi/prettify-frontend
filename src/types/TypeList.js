import React from "react";
import { Link } from "react-router-dom";
import './TypeList.css'

const TypeList = ({ types }) =>{
 
 
  return (
      <>
          {types.length > 1 ?
          <div className="TypeLists">
          {types.map(t => (
            <Link className="TypeList-link" to={`/types/${t.type}/${t.name}`} key={t.id}>
              {(t.name).trim().replace("_", " ")}
            </Link>           
          ))}
          </div>
          :
          <div className="TypeList">
            <Link className="TypeList-link" to={`/types/${types[0].type}/${types[0].name}`} key={types.id}>
              {(types[0].name).trim().replace("_", " ")}
            </Link> 
          </div>
          }
     </>
    );
}

export default  TypeList;