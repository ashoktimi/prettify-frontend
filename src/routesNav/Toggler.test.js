import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { UserProvider,  commonBeforeEach, commonAfterEach} from "../testUtils";
import { configureStore } from '@reduxjs/toolkit';
import { BrowserRouter } from "react-router-dom";
import cartReducer from '../cart/cartSlice';
import Toggler from "./Toggler";
import BrandList from "../brands/BrandList";
import CategoryList from "../categories/CategoyList";
import TagLists from "../tags/TagList";
import TypeList from "../types/TypeList";

beforeEach(commonBeforeEach)
afterEach(commonAfterEach)

describe('Toggler', () => {
  const store = configureStore({
    reducer: {
      cart: cartReducer
    }
  });

  test("renders without crashing", function () {
    render(
     <BrowserRouter>
      <Provider store={store}>
        <UserProvider>
          <Toggler />
        </UserProvider>
      </Provider>
    </BrowserRouter>
    );
  });
  
  test("matches snapshot", function () {
    const { asFragment } = render(
      <BrowserRouter>
        <Provider store={store}>
          <UserProvider>
            <Toggler />
          </UserProvider>
        </Provider>
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('BrandList component renders correctly', () => {
    const { container } = render(
      <BrowserRouter>
        <UserProvider>
          <BrandList />
        </UserProvider>
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });
  
  test('CategoryList component renders correctly', () => {
    const { container } = render(
      <BrowserRouter>
          <UserProvider>
            <CategoryList />
          </UserProvider>
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });
  
  test('TypeList component renders correctly', () => {
    const { container } = render(
      <BrowserRouter>
        <UserProvider>
          <TypeList />
        </UserProvider>
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });
  
  test('TagLists component renders correctly', () => {
    const { container } = render(
      <BrowserRouter>
        <UserProvider>
          <TagLists />
        </UserProvider>
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });
});