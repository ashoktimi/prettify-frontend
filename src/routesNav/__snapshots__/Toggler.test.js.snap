// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggler BrandList component renders correctly 1`] = `<div />`;

exports[`Toggler CategoryList component renders correctly 1`] = `<div />`;

exports[`Toggler TagLists component renders correctly 1`] = `<div />`;

exports[`Toggler TypeList component renders correctly 1`] = `
<div>
  <div
    class="LoadingSpinner"
  >
    Loading ...
  </div>
</div>
`;

exports[`Toggler matches snapshot 1`] = `
<DocumentFragment>
  <nav
    class="navbar navbar-expand-bg navbar-light"
    style="padding: 0px;"
  >
    <div
      class="container-fluid"
    >
      <button
        aria-controls="offcanvasNavbar-expand"
        aria-label="Toggle navigation"
        class="navbar-toggler collapsed"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="NavBar-name-div"
      >
        <a
          class="NavBar-name"
          href="/products"
        >
          Prettify
        </a>
      </div>
      <div
        aria-labelledby="offcanvasNavbarLabel-expand"
        class="offcanvas offcanvas-start"
        id="offcanvasNavbar-expand"
      >
        <h3
          class="Toggler-h3"
        >
          Please login to see.
        </h3>
      </div>
    </div>
  </nav>
</DocumentFragment>
`;
